### PR TITLE
op-chain-ops: incremental chain deployments [super experimental draft]

### LINKED ISSUE #11281: Deploy-chain genesis generation MVP
Implement MVP of https://github.com/ethereum-optimism/design-docs/pull/52
This sets up the genesis of L1 and L2, supporting multiple L2s, without in-depth configuration or more step-wise deployments.

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Early draft, see design https://github.com/ethereum-optimism/design-docs/pull/52

**Tests**

TODO

**Additional context**

TODO

**Metadata**

Fixes https://github.com/ethereum-optimism/optimism/issues/11281

### CODE REVIEW

File: op-chain-ops/setup/deploy/l1.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+// DevL1 defines how to create a initial L1 chain state for dev purposes
+type DevL1 struct {
+	Args genesis.DevL1DeployConfig
+
+	Addresses struct{} // No dependencies
+}
+
+func (cfg *DevL1) ScriptTarget() string {
```
Comment (tynes) [inline op-chain-ops/setup/deploy/l1.go:14]: Are you thinking of making this an interface?

File: op-chain-ops/setup/deploy/l2_genesis.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+func (cfg *FullL2) ScriptTarget() string {
+	return "L2Genesis.s.sol"
+}
+
+func (cfg *FullL2) ScriptSig() string {
+	return "l2Genesis"
+}
+
+func (cfg *FullL2) ScriptDependencies() []string {
```
Comment (tynes) [inline op-chain-ops/setup/deploy/l2_genesis.go:23]: I am a bit worried about having to list contract names here as it creates this leaky abstraction between solidity and go. Could the `ScriptDependencies` be other Go `Deployment`s (not sure what interface name is) ?

**Comment by tynes** [comment]
Perhaps we could implement less in go and have less than perfect caching now but then work with the foundry team to improve caching in the future. This prevents the need to reimplement & maintain a bunch of work that foundry should be able to do.

We could use `forge tree` plus regex for `vm.getCode` if we need to populate a deps list rather than hardcoding them in Go. What I want to avoid is implicit dependencies on modifying far away parts of the codebase for correctness

**Comment by protolambda** [comment]
Closed the design doc in favor of the Go forge script approach to integration into op-e2e. Closing this draft also now.

**Pull request closed without merging**